src/web/animasjon/animasjon.md

```diff
@@ -0,0 +1,447 @@
+
+</style>
+</head>
+<body>
+    <div id="boks"></div>
+</body>
+</html>
+
+```
+
+__La oss se nærmere på koden over:__
+
+Vi har en `<div>` med ID `boks`, den er 50x50px med blå bakgrunnsfarge. Posisjonen er `relative` som vil si at vi har muligheter for å flytte på den.
+
+`Animation`-attributtet:
+- `name`: frem_og_tilbake
+- `duration`: 2s (sekunder)
+- `timing-function`: blank, siden vi ikke trenger den
+- `delay`: 0s
+- `iteration-count`: infinite (uendelig, så den vil ikke stoppe)
+- `direction`: alternate (for at den skal gå frem og tilbake)
+- `fill-mode`: both (samme stil på start og slutt)
+
+`@keyframes frem_og_tilbake` er det vi bruker for å spesifisere hva som skal skje under animasjonen. I dette tilfellet har vi satt navnet til animasjonen til `frem_og_tilbake`, vi bruker derfor `@keyframes frem_og_tilbake` for å faktisk animere animasjonen. Innen for `@keyframes` kan vi nå spesifisere hva vi vil at animasjonen skal gjøre, over har vi 2 faser, en start og en slutt. Siden det står `0%` betyr dette når animasjonen starter og `100%` betyr når animasjonen er ferdig. Derfor vil animasjonen vår starte i `left: 0px; top: 0px;` og ende i `left: 100px; top 0px;`. 
+
+__NB!__ Verdiene i `animation`-attributtet kan også skrives som et eget attributt på denne måten: 
+```css
+#boks{
+    animation-name: frem_og_tilbake;
+    animation-duration: 2s;
+    animation-delay: 0s;
+    ...
+}
+```
+
+## Utfordring {.challenge}
+- Skriv koden inn i din favoritt teksteditor, lagre det som en `.html`-fil og gjør oppgavene under
+- Få animasjonen til å bytte farge fra blå til rød underveis.
+- Klarer du å få boksen til å flytte seg nedover og oppover?
+- Prøv å få boksen til å bevege seg i en firkant
+
+
+# Steg 2: @keyframes {.activity}
+
+La oss nå se litt nærmere på `@keyframes` og hva vi trenger å gjøre for å få animasjon.
+
+`@keyframes` forteller CSS at vi skal starte en animasjon, måten vi kan animere på er å legge inn CSS i `@keyframes`.
+
+Her kommer noen eksempler:
+<style>
+    
+    #diagonalt{
+        height: 50px;
+        width: 50px;
+        background-color: green;
+        position: relative;
+        
+        animation: diagonalt 2s 0s infinite;
+    }
+    
+    
+    @keyframes diagonalt{
+        from{top: 0px; left: 0px; }
+        to{top: 100px; left: 100px; }  
+    }
+
+</style>
+
+<div id="diagonalt"></div>
+<br>
+<br>
+<br>
+<br>
+<br>
+
+```css
+@keyframes diagonalt{
+    from{top: 0px; left: 0px; }
+    to{top: 100px; left: 100px; }
+}
+```
+Dette eksempelet får et objekt til å gå diagonalt siden det starter på `top: 0px; left: 0px` og ender på `top: 100px; left: 100px;`. 
+
+<style>
+    
+    #ned{
+        height: 50px;
+        width: 50px;
+        background-color: red;
+        position: relative;
+        
+        animation: ned 2s 0s infinite;
+    }
+    
+    
+    @keyframes ned{
+        from{top: 0px;}
+        to{top: 100px;}  
+    }
+
+</style>
+
+<div id="ned"></div>
+<br>
+<br>
+<br>
+<br>
+<br>
+
```

Comment:
Det der er bare for å få mellomrom i oppgaveteksten sånn at animasjonen ikke slik over teksten. Legger det inn i CSS-en slik at det blir penere kode. :)